src/azure_devtools/scenario_tests/utilities.py

```diff
@@ -34,3 +34,18 @@ def get_sha1_hash(file_path):
             sha1.update(data)
 
     return sha1.hexdigest()
+
+
+def is_text_payload(entity):
+    text_content_list = ['application/json', 'application/xml', 'text/html']
+
+    # 'headers' is a field of 'request', but it is a dict-key in 'response'
+    headers = getattr(entity, 'headers', None)
+    if headers is None:
+        headers = entity.get('headers')
+
+    if headers:
+        # content-type is an array from response, convert to string first for keyword match
+        content_type = str(headers.get('content-type', None))
```

Comment:
```python
content_type = str(headers.get('content-type', None)).split(";")[0].lower()
```
since content-type could be:
```
Application/JSON; charset=utf-8
```

Reply:
The partial match is intentional,  as i don't know whether `charset` could be other value.